test/integration/features/basic.feature

```diff
@@ -0,0 +1,20 @@
+# 
+Feature: Bringup
+  In order to use MiniShift
+  As a user
+  I need to be able to bring up a test environment
+
+  Scenario: Docker environment variables
+    Given a "developer"
+     When executing "minishift start --docker-env=FOO=BAR --docker-env=BAZ=BAT"
+      And should be running
+     Then should have a valid IP address
+     Then executing "minishift ssh echo hello" should contain
+      """
+      hello
+      """
+     Then executing "minishift ssh cat /var/lib/boot2docker/profile" should contain
```

Comment:
> Not sure what you mean with the verification of the content of a file, if it has been changed. 

This was just an example of an additional test condition which one could have after executing a command. For example:

     When executing "minishift config set memory 2048" 
     Then stdout of "minishift config set memory 2048" should match
       """
         Memory set to 2048
       """
     And exit status of "minishift config set memory 2048" is 0
     And file '$MINISHIFT_HOME/config/config.json' should contain
     """
    \"memory\": 4096
    """

I am most interested in how this reads and how it communicates the actual test in terms of preconditions and following assertions. I get that the 'When', 'Then' part is not enforced in the step definitions.

> as there is not distinguishing between 'When', 'Then' or 'And' and no relation can be given to them.

Sure

> The same goes for the described example. This means that the 'Then' step needs to register an index, based on the command given, because a successive step would need this to know which 'stderr' or 'stdout' is requested. 

Right, so in the example the command itself is the key/index

>I have no problem with this... but I can not enforce the usage of 'Then' and 'And', they have the same meaning in good.

Sure. I don't think you can do that in Cucumber in general.

> Which means that the index is set by the last executed command (or only the last command is kept on record). This way, no association needs to be kept. WDYT?

Could work. It would break however, if your precondition consists of two commands (example might be contrived):

     When executing "minishift config set memory 2048" 
     And executing "minishift config unset memory" 
     And file '$MINISHIFT_HOME/config/config.json' should be empty

This case might not be so common though. 

> as this makes the Gherkin VERY stringent and inflexible.

Why? Because you repeat the command? It is basically just the index you have been talking about. If one wanted to work around that one could do something like:

     When executing "minishift config set memory 2048" as "set-command"
     Then exit status of "set-command" is 0

Effectively allowing to specify your own index. Not sure whether this is much better. 

I could go either way. Implicit index in a way that just the last command is kept, command itself is key/index or even the ability to provide key/index. WDYT?